src/main/java/com/monumental/services/AsyncJobService.java

```diff
@@ -0,0 +1,29 @@
+package com.monumental.services;
+
+import com.monumental.util.async.AsyncJob;
+import org.springframework.stereotype.Service;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This service maintains an internal job queue in order to report progress and results of AsyncJobs as they run
+ */
+@Service
+public class AsyncJobService {
+
+    private Map<Integer, AsyncJob> jobs = new HashMap<>();
```

Comment:
Do jobs ever get removed from this `Map`?

Reply:
No, only when the server restarts, since this is only stored in memory